components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -318,4 +321,54 @@ void PageContentExtractor::GetSearchSummarizerKey(
   std::move(callback).Run(element.GetAttribute("content").Utf8());
 }
 
+void PageContentExtractor::ValidateOpenLeoButtonNonce(
+    const std::string& nonce,
+    mojom::PageContentExtractor::ValidateOpenLeoButtonNonceCallback callback) {
+  if (nonce.empty()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  auto element = render_frame()->GetWebFrame()->GetDocument().GetElementById(
+      "continue-with-leo");
+  if (element.IsNull() || !element.HasHTMLTagName("a")) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  GURL url(element.GetAttribute("href").Utf8());
```

Comment:
I believe it's covered in test case, it would be empty, hence an invalid URL.
This function correctly runs callback with false without any unexpected behavior or crashes.